akka-remote/src/main/scala/akka/remote/artery/Codecs.scala

```diff
@@ -47,6 +47,11 @@ private[remote] class Encoder(
         val outboundEnvelope = grab(in)
         val envelope = bufferPool.acquire()
 
+        // FIXME: OMG race between setting the version, and using the table!!!!
```

Comment:
Just noticed this race this friday, scenario is:

- incoming messages are concurrent to outgoing ones
- incoming message may be table advertisement
- which swaps the table in `Outgoing*Compression` for the new one (`n+1`)
- by itself it does so atomically,
- race: however here we store the compression table version separately from actually using it (storing the refs / manifests etc).
  - so there is a slight race IF the table is swapped right between us setting the version `n` here [then the table being swapped to `n+1`] and then we use the `n+1` version to compressions the compressions (which the receiving end will fail to read, since the encoding could be completely different, and it picks the table based on the version Int).

A solution would be to `getTable => use it to set and compress things`